### ISSUE TITLE
bazel-skylib is failing at BAZEL@HEAD

### ISSUE THREAD

**Issue by sgowroji** [bazelbuild/bazel-skylib#459]
https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/3301#018a8729-1c10-4eb0-84a6-fbf82bc08eb8

Platform : Macos, Ubunut, Windows

Logs
```
ERROR: /private/var/tmp/_bazel_buildkite/f2ac65e2a4189f8c9b21d9a2462fd49f/external/io_bazel_rules_go/go/private/context.bzl:556:23: Use of Starlark transition without allowlist attribute '_allowlist_function_transition'. See Starlark transitions documentation for details and usage: @io_bazel_rules_go//go/private:context.bzl NORMAL
(04:17:18) ERROR: Skipping '@bazel_skylib_gazelle_plugin//...': error loading package under directory '': error loading package '@bazel_skylib_gazelle_plugin//bzl': at /private/var/tmp/_bazel_buildkite/f2ac65e2a4189f8c9b21d9a2462fd49f/external/io_bazel_rules_go/go/def.bzl:26:5: initialization of module 'go/private/context.bzl' failed
(04:17:18) ERROR: error loading package under directory '': error loading package '@bazel_skylib_gazelle_plugin//bzl': at /private/var/tmp/_bazel_buildkite/f2ac65e2a4189f8c9b21d9a2462fd49f/external/io_bazel_rules_go/go/def.bzl:26:5: initialization of module 'go/private/context.bzl' failed

```
Steps:
```
git clone https://github.com/bazelbuild/bazel-skylib.git && cd bazel-skylib
git reset 7760d83db7e57961b2d865f75788b3b5f5db6ab8 --hard
export USE_BAZEL_VERSION=07e0d316a345a3cb2593f98525320590bbc56e30
bazel build //... @bazel_skylib_gazelle_plugin//...
```
Culprit : https://github.com/bazelbuild/bazel/commit/10a46b1e85f0b5bf8564c125dd02cd4e70017452

CC Greenteam @meteorcloudy

**Comment by fmeum** [comment]
@comius I think we should roll back https://github.com/bazelbuild/bazel/commit/10a46b1e85f0b5bf8564c125dd02cd4e70017452 and coordinate fixes to rulesets first. To fix skylib, we would first need to fix rules_go, then release, then update skylib, ... This would keep downstream broken for a while.

**Comment by sgowroji** [comment]
This Issue is Fixed with [6bf7bae](https://github.com/bazelbuild/bazel-skylib/commit/6bf7bae2f4882afcc82f48aebcd39ad467547b3c) . We can close it.
